docs/user-guide/rug/generators.md

```diff
@@ -158,23 +280,204 @@ the files copied from the generator project.  Using this object, you
+ Then the name of the application tests file is changed
+ Then the name of the integration tests file is changed
+ Then the name of the web integration tests file is changed
+
+ Then the name of the class in the application file is changed
+ Then the name of the class in the configuration file is changed
+ Then the name of the class in the application tests file is changed
+ Then the name of the class in the integration tests file is changed
+ Then the name of the class in the web integration tests is changed
+```
+
+Implemented by the steps in `.atomist/tests/Steps.ts` file:
+
+```typescript linenums="1"
+import { Given, When, Then, Result, ProjectScenarioWorld } from "@atomist/rug/test/Core";
+import { Project } from "@atomist/rug/model/Project";
+
+Given("an empty project", p => {})
+
+When("running the Spring Boot Service generator", (p: Project, world: ProjectScenarioWorld) => {
+  let generator = world.generator("NewSpringBootService");
+  world.generateWith(generator, {"service_class_name": "CalendarService"});
+})
+
+Then("the name of the application file is changed", (p: Project) => 
+    p.fileExists("src/main/java/com/company/CalendarServiceApplication.java")
+)
+Then("the name of the configuration file is changed", (p: Project) => 
+    p.fileExists("src/main/java/com/company/CalendarServiceConfiguration.java")
+)
+Then("the name of the application tests file is changed", (p: Project) => 
+    p.fileExists("src/test/java/com/company/CalendarServiceApplicationTests.java")
+)
+Then("the name of the integration tests file is changed", (p: Project) => 
+    p.fileExists("src/test/java/com/company/CalendarServiceOutOfContainerIntegrationTests.java")
+)
+Then("the name of the web integration tests file is changed", (p: Project) => 
+    p.fileExists("src/test/java/com/company/CalendarServiceWebIntegrationTests.java")
+)
+
+Then("the name of the class in the application file is changed", (p: Project) => 
+    p.findFile("src/main/java/com/company/CalendarServiceApplication.java").contains("CalendarServiceApplication")
+)
+Then("the name of the class in the configuration file is changed", (p: Project) => 
+    p.findFile("src/main/java/com/company/CalendarServiceConfiguration.java").contains("CalendarServiceConfiguration")
+)
+Then("the name of the class in the application tests file is changed", (p: Project) => 
+    p.findFile("src/test/java/com/company/CalendarServiceApplicationTests.java").contains("CalendarServiceApplicationTests")
+)
+Then("the name of the class in the integration tests file is changed", (p: Project) => 
+    p.findFile("src/test/java/com/company/CalendarServiceOutOfContainerIntegrationTests.java").contains("CalendarServiceOutOfContainerIntegrationTests")
+)
+Then("the name of the class in the web integration tests is changed", (p: Project) => 
+    p.findFile("src/test/java/com/company/CalendarServiceWebIntegrationTests.java").contains("CalendarServiceWebIntegrationTests")
+)
+```
```

Comment:
I can't really comment on this. I have not made my self familiar with the Gherkin stuff.